daemon/list.go

```diff
@@ -393,15 +397,24 @@ func (daemon *Daemon) transformContainer(container *container.Container, ctx *li
 // Volumes lists known volumes, using the filter to restrict the range
 // of volumes returned.
 func (daemon *Daemon) Volumes(filter string) ([]*types.Volume, error) {
-	var volumesOut []*types.Volume
+	var (
+		volumesOut []*types.Volume
+		filterUsed = false
+	)
 	volFilters, err := filters.FromParam(filter)
 	if err != nil {
 		return nil, err
 	}
-
-	filterUsed := volFilters.Include("dangling") &&
-		(volFilters.ExactMatch("dangling", "true") || volFilters.ExactMatch("dangling", "1"))
-
+	if err := volFilters.Validate(acceptedVolumeFilterTags); err != nil {
+		return nil, err
+	}
+	if volFilters.Include("dangling") {
+		if volFilters.ExactMatch("dangling", "true") {
+			filterUsed = true
+		} else if !volFilters.ExactMatch("dangling", "false") {
```

Comment:
It follows same logic as docker images, and the Validate only check allowed name